myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java

```diff
@@ -110,28 +96,28 @@ public NMTaskFactoryImpl(MyriadConfiguration cfg, TaskUtils taskUtils, ExecutorC
         }
 
         Preconditions.checkState(allAvailablePorts.size() >= NMPorts.expectedNumPorts(), "Not enough ports in offer");
-        
+
         while (ports.size() < NMPorts.expectedNumPorts()) {
           int portIndex = rand.nextInt(allAvailablePorts.size());
           ports.add(allAvailablePorts.get(portIndex));
           allAvailablePorts.remove(portIndex);
-        }        
+        }
       }
       return ports;
     }
-    
+
     //Utility function to get the first NMPorts.expectedNumPorts number of ports of an offer
     @VisibleForTesting
     protected static NMPorts getPorts(Offer offer) {
       HashSet<Long> ports = new HashSet<>();
       for (Resource resource : offer.getResourcesList()) {
-        if (resource.getName().equals("ports")) {
+        if (resource.getName().equals("ports") && (!resource.hasRole() || resource.getRole().equals("*"))) {
```

Comment:
Oh.. now I see that you mentioned the ports are obtained from * role and not from the framework role. Can you please open a JIRA to track this for future?